ci/build_cpp.sh

```diff
@@ -17,7 +17,28 @@ rapids-print-env
 
 rapids-logger "Begin C++ and Python builds"
 
-rapids-conda-retry mambabuild \
-  conda/recipes/ucxx
+# TODO: Upstream this to the image.
+mamba install rattler-build -c conda-forge
+
+# Notes on the comments in the command below (some things like file renamings
+# should be done before merging):
+# - rattler-build uses recipe.yaml by default, not meta.yaml.
+# - rattler-build uses variants.yaml by default, not conda_build_config.yaml
+# - rattler-build does not respect .condarc, so channels and the output dir
+#   must be explicitly specified
+# - The multi-output cache is currently an experimental feature.
+#   (https://prefix-dev.github.io/rattler-build/dev/multiple_output_cache/)
+# - By default rattler-build adds a timestamp that defeats sccache caching,
+#   which --no-build-id turns off
+RAPIDS_PACKAGE_VERSION=$(rapids-generate-version) rattler-build build \
+    --recipe conda/recipes/ucxx/meta.yaml \
+    --variant-config conda/recipes/ucxx/conda_build_config.yaml \
+    -c rapidsai-nightly -c conda-forge \
```

Comment:
We'll have to use `rapids-is-release-build` to switch between `rapidsai` and `rapidsai-nightly` or something similar.